src/sentry/release_health/metrics.py

```diff
@@ -615,59 +631,49 @@ def check_has_health_data(
         else:
             project_ids = projects_list  # type: ignore
 
-        org_id = self._get_org_id(project_ids)
+        projects, org_id = self._get_projects_and_org_id(project_ids)
 
-        where_clause = [
-            Condition(Column("org_id"), Op.EQ, org_id),
-            Condition(Column("project_id"), Op.IN, project_ids),
-            Condition(
-                Column("metric_id"),
-                Op.EQ,
-                resolve(USE_CASE_ID, org_id, SessionMRI.SESSION.value),
-            ),
-            Condition(Column("timestamp"), Op.GTE, start),
-            Condition(Column("timestamp"), Op.LT, now),
+        select = [MetricField(metric_mri=SessionMRI.SESSION.value, alias="value", op="sum")]
+
+        where_clause = []
+        groupby = [
+            MetricGroupByField(field="project_id"),
         ]
 
         if includes_releases:
-            releases = [x[1] for x in projects_list]  # type: ignore
-            release_column_name = resolve_tag_key(USE_CASE_ID, org_id, "release")
-            releases_ids = resolve_many_weak(USE_CASE_ID, org_id, releases)
-            where_clause.append(Condition(Column(release_column_name), Op.IN, releases_ids))
-            column_names = ["project_id", release_column_name]
-
-        else:
-            column_names = ["project_id"]
-
-        def extract_row_info_func(
-            include_releases: bool,
-        ) -> Callable[[Mapping[str, Union[int, str]]], ProjectOrRelease]:
-            def f(row: Mapping[str, Union[int, str]]) -> ProjectOrRelease:
-                if include_releases:
-                    return row["project_id"], reverse_resolve(USE_CASE_ID, org_id, row.get(release_column_name))  # type: ignore
-                else:
-                    return row["project_id"]  # type: ignore
-
-            return f
-
-        extract_row_info = extract_row_info_func(includes_releases)
-
-        query_cols = [Column(column_name) for column_name in column_names]
-        group_by_clause = query_cols
-
-        query = Query(
-            match=Entity(EntityKey.MetricsCounters.value),
-            select=query_cols,
-            where=where_clause,
-            groupby=group_by_clause,
+            where_clause.append(filter_releases_by_project_release(projects_list))  # type: ignore
+            groupby.append(MetricGroupByField(field="release"))
+
+        query = MetricsQuery(
+            org_id=org_id,
+            project_ids=project_ids,
+            select=select,
+            start=start,
+            end=now,
             granularity=Granularity(24 * 60 * 60),  # daily
```

Comment:
I would prefer if we generalize common granularities in variables like `DAY`, `HOUR`...

Reply:
Yep, I think we are already doing it in other places.